src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -463,23 +475,37 @@ private static bool TryUnfoldITestDataSource(ITestDataSource dataSource, TestDat
                 // Duplicated display name so bail out. Caller will handle adding the original test.
                 return false;
             }
-#pragma warning restore CS0618 // Type or member is obsolete
 
-            try
+            TestDataIdentifierStrategy currentDataSourceIdentifierStrategy = (dataSource as ITestDataIdentifierStrategy)?.IdentifierStrategy ?? TestDataIdentifierStrategy.Auto;
+            if (currentDataSourceIdentifierStrategy == TestDataIdentifierStrategy.DataIndex
+                || (currentDataSourceIdentifierStrategy == TestDataIdentifierStrategy.Auto && dataSourcesIdentifierStrategy == TestDataIdentifierStrategy.DataIndex))
             {
-                discoveredTest.TestMethod.SerializedData = DataSerializationHelper.Serialize(d);
+                discoveredTest.TestMethod.SerializedData = new string[3]
```

Comment:
I am reusing the serialized data property as it's not used and allows me to avoid adding more properties to the TestCase object as it has some perf impact. This isn't perfect but as far as I can tell, there is no way for the serialized data to be serialized this way as we prefix with the data type.

Reply:
Fine for me to reuse. At least until we have a strong reason not to reuse.